core/chains/evm/config/chain_scoped_gas_estimator.go

```diff
@@ -137,10 +137,12 @@ func (d *daOracleConfig) OracleAddress() *types.EIP55Address {
 	return d.c.OracleAddress
 }
 
-// CustomGasPriceCalldata returns the calldata for a custom gas price API.
+// CustomGasPriceCalldata returns the calldata for a custom gas price API. For now this is only enabled for CustomCalldata oracle types
 func (d *daOracleConfig) CustomGasPriceCalldata() string {
-	// TODO: CCIP-3710 update once custom calldata oracle is added
-	return ""
+	if d.c.OracleType != toml.CustomCalldata {
```

Comment:
this feels off, maybe we can move the Oracle constants into a child `daoracle` package, or prepend them, smt like DAOracle_CustomCalldata